web/src/util.js

```diff
@@ -1,5 +1,76 @@
-const example = x => x
+const COLOR_CLS = { num: 'red', null: 'blue', bool: 'purple', str: 'maroon',  key: 'teal', default: 'black'}
 
+const typeToClass = {
+    'boolean': COLOR_CLS.bool,
+    'number': COLOR_CLS.num,
+    'string': COLOR_CLS.str,
+    'object': COLOR_CLS.default,
+};
+
+/**
+ * Returns the stringified JSON with Syntax Highlighting
+ * @param {object | Array} myJson The Object to stringify.
+ * @returns {string} The stringified Object.
+ */
+const getStringifiedJSON = (myJson, indentation) => {
+    if (myJson === undefined) {
+        throw Error("Caution! myJson is undefined");
+    }
+    const stringifiedJSON = JSON.stringify(myJson, null, indentation)
+    let lines = stringifiedJSON.split('\n')
+    lines = lines.map(line => {
+        const end = line.endsWith(',') ? ',' : ''
+        line = end === ',' ? line.substring(0, line.length-1) : line
+        if (['{', '[', ']', '}'].includes(line.trim())) return line + end //To avoid the opening/closing brackets in further code
+        const keyVal = line.split(':')
+        if (keyVal.length === 1) {
+            const type = getTypeOfStringValue(line)
+            const cls = (type === 'object' && line.trim() === 'null') ? COLOR_CLS.null : typeToClass[type]
+            return '<span class="' + cls +'">' + line + '</span>' + end
+        }
+        if (keyVal.length >= 2) {
+            let key = ''
+            let value = ''
+            const keyIndex = keyVal.findIndex(val => val.endsWith('"'))
```

Comment:
Das ist zwar recht tricky, was du hier machst und wird in den meisten Fällen funktionieren, aber ich könnte dir immer Beispiele finden, wo dein Code failed. Ich denke z.B. hier:
`getStringifiedJSON({"\":666": "failure"}, 2)`

Um das wirklich korrekt zu machen, müsstest du hier wie ein Compiler den String parsen, d.h. du durchläufst den String Buchstabe für Buchstabe. Kommt ein `"` beginnt ein String, kommt ein `\` würde eine Escape-Sequenz starten und du kannst das nächste Zeichen immer ignorieren (es sei denn es ist ein `u`, dann kommt eine Unicode-Zeichensequenz). Wenn du nicht im Escape-Modus bist und es folgt ein `"` ist der String dann zuende und du hast ein JSON-String-Token gefunden. 

Normalerweise würde man hier in die Spec von JSON schauen (https://www.ietf.org/rfc/rfc4627.txt unter "2.5 Strings") ...die sieht mir nicht sonderlich kompliziert aus, aber evt. gibt es auch schon Regexp die das extrahieren.